lib-injection/sitecustomize.py

```diff
@@ -2,65 +2,238 @@
 This module when included on the PYTHONPATH will update the PYTHONPATH to point to a directory
 containing the ddtrace package compatible with the current Python version and platform.
 """
+
+from __future__ import print_function  # noqa: E402
+
+from collections import namedtuple
+import csv
+import json
 import os
+import platform
+import re
+import subprocess
 import sys
 import time
+from typing import Tuple
+
+
+Version = namedtuple("Version", ["version", "constraint"])
+
+
+def parse_version(version: str) -> Tuple:
+    constraint_idx = re.search(r"\d", version).start()
+    numeric = version[constraint_idx:]
+    constraint = version[:constraint_idx]
+    parsed_version = tuple(int(re.sub("[^0-9]", "", p)) for p in numeric.split("."))
+    return Version(parsed_version, constraint)
+
+
+RUNTIMES_ALLOW_LIST = {
+    "cpython": {"min": parse_version("3.7"), "max": parse_version("3.12")},
+}
+
+FORCE_INJECT = os.environ.get("DD_INJECT_FORCE", "").lower() in (
+    "true",
+    "1",
+    "t",
+)
+FORWARDER_EXECUTABLE = os.environ.get("DD_TELEMETRY_FORWARDER_PATH")
+TELEMETRY_ENABLED = os.environ.get("DD_INJECTION_ENABLED")
+DEBUG_MODE = os.environ.get("DD_TRACE_DEBUG", "").lower() in ("true", "1", "t")
+INSTALLED_PACKAGES = None
+PYTHON_VERSION = None
+PYTHON_RUNTIME = None
+PKGS_ALLOW_LIST = None
+
+
+def build_installed_pkgs():
+    if sys.version_info >= (3, 8):
+        from importlib import metadata as importlib_metadata
+
+        installed_packages = {pkg.metadata["Name"]: pkg.version for pkg in importlib_metadata.distributions()}
+    else:
+        import pkg_resources
+
+        installed_packages = {pkg.key: pkg.version for pkg in pkg_resources.working_set}
+    return installed_packages
+
+
+def build_min_pkgs():
+    min_pkgs = dict()
+    with open("../datadog-lib/min_compatible_versions.csv", "r") as csvfile:
```

Comment:
I think this path will have to change? Looking at how the install is done by the agent, it puts the `lib-injection/sitecustomize` file in `/opt/datadog/apm/library/python` and then the various tracer versions exist inside`/opt/datadog/apm/library/python/ddtrace_pkgs`? Perhaps what we want to do is just change how the install is built so that the min_compatible_versions.csv file can also just sit in `/opt/datadog/apm/library/python` with the `lib-injection/sitecustomize`?

Reply:
Ah wait this is incorrect. I'm not sure what the path should be...

Reply:
```suggestion
    with open("./min_compatible_versions.csv", "r") as csvfile:
```
I think we should just keep this file in the same directory as the sitecustomize file, since file paths can change depending on if it's host/docker or Kubernetes

Reply:
@ZStriker19 could you point me to the different bits of code that put the files that sitecustomize depends on into their intended locations? I'm aware of `copy-libs.sh` but none beyond that.

Reply:
For OCI images: https://github.com/DataDog/dd-trace-py/blob/main/.gitlab/build-oci.sh

For docker images: https://github.com/DataDog/dd-trace-py/blob/main/lib-injection/Dockerfile

Reply:
I'm having a hard time figuring out how the paths in those files map to ones that are visible to sitecustomize. As a way around this, I updated things to look for the CSV file in multiple locations.